templates/plugin/ansible.env.erb

```diff
@@ -0,0 +1,13 @@
+ANSIBLE_CALLBACK_WHITELIST="<%= @callback %>"
+ANSIBLE_LOCAL_TEMP="<%= @working_dir %>"
+ANSIBLE_HOST_KEY_CHECKING="<%= @host_key_checking ? 'True' : 'False' %>"
+ANSIBLE_STDOUT_CALLBACK="<%= @stdout_callback %>"
+ANSIBLE_ROLES_PATH="<%= @roles_path.join(':') %>"
+ANSIBLE_COLLECTIONS_PATHS="<%= @collections_paths.join(':') %>"
+
+FOREMAN_URL="<%= @foreman_url %>"
+FOREMAN_SSL_CERT="<%= @foreman_ssl_cert %>"
+FOREMAN_SSL_KEY="<%= @foreman_ssl_key %>"
+FOREMAN_SSL_VERIFY="<%= @foreman_ssl_ca %>"
```

Comment:
Rather than specifying this again, should the Smart Proxy plugin code set these variables with settings from `/etc/foreman-proxy/settings.yml` where it's already present?

Actually, since everything is already present, should everything be presented as environment variables while calling the process and avoid an additional config file to manage?

That also makes the development setup easier.

Reply:
That depends on what we really want to accomplish. Originally, we just used ansible.cfg, users could edit it but their changes were clobbered on subsequent installer runs.

With my solution, we move the things we want to set into a file that gets sourced. Users can put their own changes into ansible.cfg and they will get honored. If there is a conflict, the values set by us will take precedence. Users could still edit this file with environment variables (and have them undone on subsequent installer runs).

From a users' point of view, my solution and what you're proposing is the same, sans the option to temporarily override the values we set.

On a side note, not all of the options that are being set are currently known to sp-ansible. Yes, the installer could put them into sp-ansible's config and then it could pass them on ansible itself, but that begins to feel slightly convoluted.

Reply:
One thing I like about this solution (vs "sp-ansible generates all this on the fly) is the fact that this allows the user to "run ansible as foreman would do", which is useful for debugging.

We certainly could achieve this with code within sp-ansible, but it would require quite a lot more code than this here.

That said, I don't mind the exact implementation if y'all decide to move to something else (but this one seems to work from reading the code).

Reply:
I also thought about the debugging aspect. I just don't know if that weighs up to the bigger need to manage files. Perhaps I'm looking too far ahead, but I can imagine at some point we would use containers and then it's additional work to keep it in sync.

Reply:
In the container case, I'd expect all these to be "magically" set in the container environment by the runtime (thus, shifting the generator of the correct values from the installer to whatever orchestrates the containers)

Reply:
> If there was a way through an API or a script that could be run

We could do that. It feels a bit overengineered over "just" having the installer spit out a file that can be sourced, but sure.